File: 07_02_Greedy/92_2_assign_cookies/szzng.py
# Runtime 147 ms Beats 99.78%
# Memory 15.5 MB Beats 99.15%


class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort(reverse=True)
        s.sort()
        end = min(len(s), len(g))
        if len(g) > end: g = g[-1 * end:]
        if len(s) > end: s = s[-1 * end:]
        result = 0

        for i in g:
            if len(s) == 0:
                return result

            if s[-1] >= i:
                result += 1
                s.pop()

        return result

File: 07_02_Greedy/92_2_assign_cookies/szzng.js
// Runtime 85 ms Beats 92.44%
// Memory 44.5 MB Beats 70%

var findContentChildren = function(g, s) {
    g.sort((a,b)=>b-a)
    s.sort((a,b)=>b-a)
    var end = Math.min(g.length, s.length)
    if (g.length>end) g.splice(0, g.length-end)
    if (s.length>end) s.splice(end)
    var result = 0
    var index = 0

    for (var i=0; i < end; i++){
        if (s.length ===0) return result

        if (s[index] >= g[i]){
            result+=1
            index+=1
        }
    }

    return result
};